### PR TITLE
Added Squared Hinge loss

### PR DESCRIPTION
**Pull request by atreyamaj**
## Description
Added the Squared Hinge Loss.

Referencing Issue #2478 

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

Screenshot from 2021-04-23 12-47-10

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.22.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

@rbharath @peastman @alat-rights

### CODE REVIEW

**Comment by atreyamaj** [comment]
That's weird. I have pushed a new change on my branch for this PR but it is not reflecting here. Should I close this and open it again?

The commit message should be `Changed torch.maximum to np.maximum`

@alat-rights @rbharath

**Comment by alat-rights** [comment]
Weird. Looks like a bug on GH's end. You could try doing that.

**Comment by atreyamaj** [comment]
> Weird. Looks like a bug on GH's end. You could try doing that.

I updated the docstring and tried pushing it again, the commits were added together. Seems to be working now.

**Comment by atreyamaj** [comment]
Pytorch tensors have the option of being done on the GPU, although numpy does not have the option directly (Need to make it work with CuPy). However, the tests are failing because they are not able to run the torch math functions. Should I migrate the Pytorch function definition to Numpy or keep it in torch?

File: deepchem/models/losses.py [hunk 1]
```diff
@@ -94,6 +94,32 @@ def loss(output, labels):
...
+  def _compute_tf_loss(self, output, labels):
+    import tensorflow as tf
+    output, labels = _make_tf_shapes_consistent(output, labels)
+    return tf.keras.losses.SquaredHinge(reduction='none')(labels, output)
+
+  def _create_pytorch_loss(self):
+    import torch
+
+    def loss(output, labels):
+      import numpy as np
```
Comment (peastman) [inline deepchem/models/losses.py:112]: This import can be removed.

Comment (atreyamaj) [reply deepchem/models/losses.py]: Removed!

**Comment by rbharath** [comment]
I'm seeing some strange errors here about not having `dgllife` installed. I think that's not to do with this PR? I'm not sure if something broke elsewhere in the CI

**Comment by atreyamaj** [comment]
> I'm seeing some strange errors here about not having `dgllife` installed. I think that's not to do with this PR? I'm not sure if something broke elsewhere in the CI

I haven't modified any files other than `losses.py`, `test_losses.py` and `models.rst`

I'm not sure why `dgllife` is showing errors. @rbharath

**Review by rbharath — APPROVED** [review]
Yes, you're right! I'm seeing the errors elsewhere too. This is good to merge in on my end but will leave open for a day or two in case other folks have comments

**Comment by atreyamaj** [comment]
> Yes, you're right! I'm seeing the errors elsewhere too. This is good to merge in on my end but will leave open for a day or two in case other folks have comments

Awesome, thank you.

**Review by rbharath — APPROVED** [review]
Looks good!
